lib/galaxy/datatypes/mothur.py

```diff
@@ -0,0 +1,937 @@
+                        float(line[1])
+                    except:
+                        return False
+                count += 1
+        if count > 1:
+            return True
+
+        return False
+
+
+class Quantile(Tabular):
+    file_ext = 'mothur.quan'
+    MetadataElement(name="filtered", default=False, no_value=False, optional=True, desc="Quantiles calculated using a mask", readonly=True)
+    MetadataElement(name="masked", default=False, no_value=False, optional=True, desc="Quantiles calculated using a frequency filter", readonly=True)
+
+    def __init__(self, **kwd):
+        """Quantiles for chimera analysis"""
+        Tabular.__init__(self, **kwd)
+        self.column_names = ['num', 'ten', 'twentyfive', 'fifty', 'seventyfive', 'ninetyfive', 'ninetynine']
+        self.column_types = ['int', 'float', 'float', 'float', 'float', 'float', 'float']
+
+    def sniff(self, filename):
+        """
+        Determines whether the file is a quantiles tabular format for chimera analysis
+        1	0	0	0	0	0	0
+        2       0.309198        0.309198        0.37161 0.37161 0.37161 0.37161
+        3       0.510982        0.563213        0.693529        0.858939        1.07442 1.20608
+        ...
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'mothur_datatypetest_true.mothur.quan' )
+        >>> Quantile().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'mothur_datatypetest_false.mothur.quan' )
+        >>> Quantile().sniff( fname )
+        False
+        """
+        headers = get_headers(filename, sep='\t')
+        count = 0
+        for line in headers:
+            if not line[0].startswith('@') and not line[0].startswith('#'):
+                if len(line) != 7:
+                    return False
+                try:
+                    int(line[0])
+                    float(line[1])
+                    float(line[2])
+                    float(line[3])
+                    float(line[4])
+                    float(line[5])
+                    float(line[6])
+                except:
+                    return False
+                count += 1
+        if count > 0:
+            return True
+
+        return False
+
+
+class LaneMask(Text):
+    file_ext = 'mothur.filter'
+
+    def sniff(self, filename):
+        """
+        Determines whether the file is a lane mask filter:  1 line consisting of zeros and ones.
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'mothur_datatypetest_true.mothur.filter' )
+        >>> LaneMask().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'mothur_datatypetest_false.mothur.filter' )
+        >>> LaneMask().sniff( fname )
+        False
+        """
+        headers = get_headers(filename, sep='\t')
+        if len(headers) != 1 or len(headers[0]) != 1:
+            return False
+
+        if not re.match('^[01]+$', headers[0][0]):
```

Comment:
Is a regex on a single char not overkill?

`if not headers[0] in ['0','1']`
(length of heders[0] must be one after the previous if statement)